File: src/main/java/jp/co/sunarch/apps/idprovider/model/oauth2/code/AuthorizationCodeFlow.java
package jp.co.sunarch.apps.idprovider.model.oauth2.code;

import jp.co.sunarch.apps.idprovider.model.oauth2.AuthorizationEndpoint;
import jp.co.sunarch.apps.idprovider.model.oauth2.EndpointRequestFactory;
import jp.co.sunarch.apps.idprovider.model.oauth2.TokenEndpoint;
import jp.co.sunarch.apps.idprovider.model.oauth2.AccessTokenResponse;

/**
 * @author takeshi
 */
public class AuthorizationCodeFlow implements
    AuthorizationEndpoint<AuthorizationCodeAuthorizationRequest, AuthorizationCodeAuthorizationResponse>,
    TokenEndpoint<AuthorizationCodeAccessTokenRequest, AccessTokenResponse> {

  @Override
  public EndpointRequestFactory<AuthorizationCodeAuthorizationRequest> authorizationRequestFactory() {
    // TODO
    return null;
  }

  @Override
  public EndpointRequestFactory<AuthorizationCodeAccessTokenRequest> tokenRequestFactory() {
    // TODO
    return null;
  }

  @Override
  public AuthorizationCodeAuthorizationResponse authorize(AuthorizationCodeAuthorizationRequest request) {
    // TODO
    return null;
  }

  @Override
  public AccessTokenResponse createToken(AuthorizationCodeAccessTokenRequest request) {
    // TODO
    return null;
  }

}

File: src/main/kotlin/jp/co/sunarch/apps/idprovider/model/oauth2/code/AuthorizationCodeFlow.kt
package jp.co.sunarch.apps.idprovider.model.oauth2.code

import jp.co.sunarch.apps.idprovider.model.oauth2.AuthorizationEndpoint
import jp.co.sunarch.apps.idprovider.model.oauth2.EndpointRequestFactory
import jp.co.sunarch.apps.idprovider.model.oauth2.TokenEndpoint
import jp.co.sunarch.apps.idprovider.model.oauth2.AccessTokenResponse

/**
 * @author takeshi
 */
class AuthorizationCodeFlow : AuthorizationEndpoint<AuthorizationCodeAuthorizationRequest, AuthorizationCodeAuthorizationResponse>, TokenEndpoint<AuthorizationCodeAccessTokenRequest, AccessTokenResponse> {

  override fun authorizationRequestFactory(): EndpointRequestFactory<AuthorizationCodeAuthorizationRequest>? {
    // TODO
    return null
  }

  override fun tokenRequestFactory(): EndpointRequestFactory<AuthorizationCodeAccessTokenRequest>? {
    // TODO
    return null
  }

  override fun authorize(request: AuthorizationCodeAuthorizationRequest): AuthorizationCodeAuthorizationResponse? {
    // TODO
    return null
  }

  override fun createToken(request: AuthorizationCodeAccessTokenRequest): AccessTokenResponse? {
    // TODO
    return null
  }

}
